### PR TITLE
Add repository classes for Product

### PR DESCRIPTION
**Pull request by jakubtobiasz**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 2.0
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | fulfills #15526 
| License         | MIT

### CODE REVIEW

**Review by Wojdylak — DISMISSED** [review]

File: UPGRADE-2.0.md [hunk 1]
```diff
@@ -122,6 +122,18 @@
...
+
+    * `Sylius\Bundle\ProductBundle\Doctrine\ORM\ProductAssociationRepository`
+    * `Sylius\Bundle\ProductBundle\Doctrine\ORM\ProductAssociationTypeTranslationRepository`
+    * `Sylius\Bundle\ProductBundle\Doctrine\ORM\ProductAttributeRepository`
+    * `Sylius\Bundle\ProductBundle\Doctrine\ORM\ProductAttributeTranslationRepository`
+    * `Sylius\Bundle\ProductBundle\Doctrine\ORM\ProductOptionTranslationRepository`
+    * `Sylius\Bundle\ProductBundle\Doctrine\ORM\ProductOptionValueRepository`
+    * `Sylius\Bundle\ProductBundle\Doctrine\ORM\ProductOptionValueTranslationRepository`
+    * `Sylius\Bundle\ProductBundle\Doctrine\ORM\ProductTranslationRepository`
+    * `Sylius\Bundle\ProductBundle\Doctrine\ORM\ProductVariantTranslationRepository`
```
Comment (Wojdylak) [inline UPGRADE-2.0.md:136]:
```suggestion
    * `Sylius\Bundle\ProductBundle\Doctrine\ORM\ProductVariantTranslationRepository` has been added
```

**Review by diimpp — COMMENTED** [review]
Hi, PR looks good.


1. This needs to be removed as well, since it's now specified via DI Configuration.

https://github.com/Sylius/Sylius/blob/336682edb5be6e2885bde0179816891315d89777/src/Sylius/Bundle/CoreBundle/Resources/config/app/sylius/sylius_product.yml#L24-L26

2. Is this `Core` `AttributeRepository` used anywhere? I cannot find where it's configured at quick glance
https://github.com/Sylius/Sylius/blob/1.13/src/Sylius/Bundle/CoreBundle/Doctrine/ORM/AttributeRepository.php

3. PR header mentions `User`, but there are no user related changes in the PR files.

File: UPGRADE-2.0.md (hunk 1 above)
Comment (jakubtobiasz) [reply UPGRADE-2.0.md:136]: CleanShot 2023-11-20 at 16 40 17@2x
IMO the `has been added` is pointless here, if previously it was included, we can remove it instead adding it here.

**Comment by jakubtobiasz** [comment]
@diimpp thanks for the review,
1. Done
2. I don't see any configuration for it neither, but we'll cover it later.
3. Yeah, the scope was to cover the User too, but there were nothing to do.

**Review by diimpp — APPROVED** [review]

**Comment by diimpp** [comment]
> 2\. I don't see any configuration for it neither, but we'll cover it later.

Should I create issue for follow-up?

**Review by Wojdylak — APPROVED** [review]

**Comment by Wojdylak** [comment]
Thank you, @jakubtobiasz!

**Pull request merged**